stereo/plots/paga_plot.py

```diff
@@ -0,0 +1,449 @@
+                barnesHutOptimize=True,
+                barnesHutTheta=1.2,
+                multiThreaded=False,  # NOT IMPLEMENTED
+                # Tuning
+                scalingRatio=2.0,
+                strongGravityMode=False,
+                gravity=1.0,
+                # Log
+                verbose=False,
+            )
+            if 'maxiter' in layout_kwds:
+                iterations = layout_kwds['maxiter']
+            elif 'iterations' in layout_kwds:
+                iterations = layout_kwds['iterations']
+            else:
+                iterations = 500
+            pos_list = forceatlas2.forceatlas2(
+                adjacency, pos=init_coords, iterations=iterations
+            )
+            pos = {n: [p[0], -p[1]] for n, p in enumerate(pos_list)}
+        elif layout == 'eq_tree':
+            nx_g_tree = nx.Graph(adj_tree)
+            pos = _utils.hierarchy_pos(nx_g_tree, root)
+            if len(pos) < adjacency.shape[0]:
+                raise ValueError(
+                    'This is a forest and not a single tree. '
+                    'Try another `layout`, e.g., {\'fr\'}.'
+                )
+        else:
+            # igraph layouts
+            random.seed(random_state.bytes(8))
+            g = self._get_igraph_from_adjacency(adjacency)
+            if 'rt' in layout:
+                g_tree = self._get_igraph_from_adjacency(adj_tree)
+                pos_list = g_tree.layout(
+                    layout, root=root if isinstance(root, list) else [root]
+                ).coords
+            elif layout == 'circle':
+                pos_list = g.layout(layout).coords
+            else:
+                # I don't know why this is necessary
+                # np.random.seed(random_state)
+                if init_pos is None:
+                    init_coords = random_state.random_sample(
+                        (adjacency.shape[0], 2)
+                    ).tolist()
+                else:
+                    init_pos = init_pos.copy()
+                    # this is a super-weird hack that is necessary as igraph’s
+                    # layout function seems to do some strange stuff here
+                    init_pos[:, 1] *= -1
+                    init_coords = init_pos.tolist()
+                try:
+                    pos_list = g.layout(
+                        layout, seed=init_coords, weights='weight', **layout_kwds
+                    ).coords
+                except AttributeError:  # hack for empty graphs...
+                    pos_list = g.layout(layout, seed=init_coords, **layout_kwds).coords
+            pos = {n: [p[0], -p[1]] for n, p in enumerate(pos_list)}
+        if len(pos) == 1:
+            pos[0] = (0.5, 0.5)
+        pos_array = np.array([pos[n] for count, n in enumerate(nx_g_solid)])
+        #stereo_exp_data.tl.result['paga']['pos'] = pos_array
+        return pos_array
+    
+    
+    def paga_plot(
+        self, 
+        sdata,
```

Comment:
In a class which inherits from PlotBase, you can get the StereoExpData object from attribute `self.stereo_exp_data`, don't need to input by yourself.